EIPS/eip-223.md

```diff
@@ -0,0 +1,389 @@
+     * Tokens usually opt for a value of 18, imitating the relationship between
+     * Ether and Wei. This is the value {ERC223} uses, unless {_setupDecimals} is
+     * called.
+     *
+     * NOTE: This information is only used for _display_ purposes: it in
+     * no way affects any of the arithmetic of the contract, including
+     * {IERC223-balanceOf} and {IERC223-transfer}.
+     */
+    function decimals() public view returns (uint8)
+    {
+        return _decimals;
+    }
+
+    /**
+     * @dev See {IERC223-totalSupply}.
+     */
+    function totalSupply() public view returns (uint256)
+    {
+        return _totalSupply;
+    }
+
+    
+    /**
+     * @dev Returns balance of the `_owner`.
+     *
+     * @param _owner   The address whose balance will be returned.
+     * @return balance Balance of the `_owner`.
+     */
+    function balanceOf(address _owner) public view returns (uint256)
+    {
+        return balances[_owner];
+    }
+    
+    /**
+     * @dev Transfer the specified amount of tokens to the specified address.
+     *      Invokes the `tokenFallback` function if the recipient is a contract.
+     *      The token transfer fails if the recipient is a contract
+     *      but does not implement the `tokenFallback` function
+     *      or the fallback function to receive funds.
+     *
+     * @param _to    Receiver address.
+     * @param _value Amount of tokens that will be transferred.
+     * @param _data  Transaction metadata.
+     */
+    function transfer(address _to, uint _value, bytes calldata _data) public returns (bool success)
+    {
+        // Standard function transfer similar to ERC20 transfer with no _data .
+        // Added due to backwards compatibility reasons .
+        balances[msg.sender] = balances[msg.sender] - _value;
+        balances[_to] = balances[_to] + _value;
+        if(Address.isContract(_to)) {
+            IERC223Recipient(_to).tokenReceived(msg.sender, _value, _data);
+        }
+        emit Transfer(msg.sender, _to, _value);
+        emit TransferData(_data);
+        return true;
+    }
+    
+    /**
+     * @dev Transfer the specified amount of tokens to the specified address.
+     *      This function works the same with the previous one
+     *      but doesn't contain `_data` param.
+     *      Added due to backwards compatibility reasons.
+     *
+     * @param _to    Receiver address.
+     * @param _value Amount of tokens that will be transferred.
+     */
+    function transfer(address _to, uint _value) public returns (bool success)
+    {
+        bytes memory _empty = hex"00000000";
+        balances[msg.sender] = balances[msg.sender] - _value;
+        balances[_to] = balances[_to] + _value;
+        if(Address.isContract(_to)) {
+            IERC223Recipient(_to).tokenReceived(msg.sender, _value, _empty);
+        }
+        emit Transfer(msg.sender, _to, _value);
+        emit TransferData(_empty);
+        return true;
+    }
+}
+
+```
+
+
+## Copyright
```

Comment:
The copyright waiver is not optional:

```suggestion
## Copyright

Copyright and related rights waived via [CC0](../LICENSE.md).
```

Reply:
Yea, it should be there. No idea where it got lost.